addons/mail/models/discuss/discuss_channel.py

```diff
@@ -672,10 +672,11 @@ def set_message_pin(self, message_id, pinned):
                 </div>
             '''
             notification = Markup(notification_text) % {
-                'user_pinned_a_message_to_this_channel': Markup('<a href="#" data-oe-type="highlight" data-oe-id="%s">%s</a>') % (
-                    message_id,
-                    _('%(user_name)s pinned a message to this channel.', user_name=self.env.user.display_name),
-                ),
+                'user_pinned_a_message_to_this_channel': Markup(_(
+                        "%(user_name)s pinned a <a href='#' data-oe-type='highlight' data-oe-id=%(message_id)s>message</a> to this channel.",
+                        user_name=self.env.user.display_name,
+                        message_id=message_id
+                )),
```

Comment:
I don't understand what is this change, but in it's current state, it is allowing XSS, as escape used to be done through markup `%s` but now it is directly injected raw from the translation value.

Reply:
Not directly related to this dev, someone recently changed this part of the code and the whole text is appearing blue (because it is inside an anchor tag). Juliette asked me to only have the word "message" highlithed as it was before.
Could you help me understand how this little change introduced XSS vulnerabilities ?